mangaki/mangaki/utils/ratingsmatrix.py

```diff
@@ -0,0 +1,42 @@
+from scipy.sparse import csc_matrix
+from mangaki.utils.values import rating_values
+import pandas
+
+
+class RatingsMatrix():
+
+    def __init__(self, qs=None, fname=None):
+        user_list, item_list, data = [], [], []
+        if fname is None:
+            if qs is None:
+                raise ValueError('one of fname or qs must be non None')
+            else:
+                content = qs
+                for user_id, item_id, rating in content:
+                    user_list.append(user_id)
+                    item_list.append(item_id)
+                    data.append(rating_values[rating])
+        else:
+            content = pandas.read_csv(fname,
+                                      header=None).as_matrix()
+            for user_id, item_id, rating in content:
+                user_list.append(user_id)
+                item_list.append(item_id)
+                data.append(rating_values[rating])
+        user_set = set(user_list)
+        item_set = set(item_list)
+        user_dict = {v: k for k, v in enumerate(user_set)}
```

Comment:
Ce n'est pas `dict(enumerate(user_set))` parceque ça, ça serait `{k: v for k, v in enumerate(user_set)}` alors qu'ici on a `{v: k for k, v in enumerate(user_set)}` (inversion de `k` et `v`.